src/Context.tsx

```diff
@@ -7,7 +7,7 @@ const PuzzleContext = createContext<PuzzlesContextInterface>(null!);
 const PuzzleProvider: React.FC = ({children}) => {
 	const [puzzles, setPuzzles] = useState([])
 	const [loggedIn, setLoggedIn] = useState(false)
-	const [user, setUser] = useState({})
+	const [user, setUser] = useState<UserObjectInterface>(null!)
```

Comment:
There's a bit of an anti pattern happening here. The purpose of the (!) operator in TS is to explicitly tell your editor that this variable will never be null. So adding this onto an explicit null here is a bit whacky and I have no idea why it solves your problem lol... But I wouldn't recommend it! 

The purpose of that operator is for example, maybe you're mapping over an array that is declared higher up. If you try `myArray.map(blah =>` your editor will take issue with the fact that myArray might possibly be null (maybe it's coming back from an API request) but if you say `myArray!.map` to tell your editor, don't worry about showing me the "possibly undefined error, this thing will always be defined here, I promise". 

I think what would work better for you here is 
```
const [user, setUser] = useState({} as UserObjectInterface);
```

The "as" is called a [type assertion](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions) and lets you tell the compiler that your `user` object will be in this shape, so if it's not, let me know there's a problem. 

You can do the same for the PuzzleContext variable above.

Reply:
Another thing worth mentioning is standard convention is to type interfaces as `IMyInterface`, so in this case `IUserObject` and then the person reading it always knows it's an interface and not a generic type. Not super important, just wanted to point that out.